docs/guides/other-guides/path-based-routing-and-policy-decentralization-with-cloud-endpoints.mdx

```diff
@@ -0,0 +1,260 @@
+    -H "Authorization: Bearer ${NGROK_API_KEY}" \
+    -H "Content-Type: application/json" \
+    -H "Ngrok-Version: 2" \
+    -d @- <<BODY
+    {
+        "name":"${NGROK_SUBDOMAIN}.ngrok.app",
+    }
+BODY
+```
+
+- Replace or set `NGROK_API_KEY` to your ngrok API key.
+- Replace or set `NGROK_SUBDOMAIN` as the subdomain you'd like to use for this guide.
+
+After running, you should see the following:
+
+```json
+200 OK
+{
+   "id":"rd_2MT5Bqt0UzU0mFQ0zr8m1UQWCfm",
+   ...
+}
+```
+
+</TabItem>
+<TabItem value="dash" label="ngrok Dashboard">
+You can reserve a subdomain via [ngrok dashboard](https://dashboard.ngrok.com/domains/new). 
+After successfully reserving a domain, you should see it listed in the domains table.
+</TabItem>
+</Tabs>
+
+When you have completed this step, you can move on to the next step.
+
+## **Step 4** — Create a Traffic Policy for your Cloud Endpoint
+
+Now, we'll continue and create a traffic policy for the cloud endpoint.
+All incoming traffic will be preprocessed by this traffic policy, before
+being forwarded to the internal endpoints which may or may not have their
+own independent traffic policies. This separation allows for granularity and
+full control over how traffic is handled.
+
+<ConfigExample
+	snippetText={null}
+	showLineNumbers={true}
+	config={{
+		on_http_request: [
+			{
+				expressions: ["req.url.path.startsWith('/api/')"],
+				actions: [
+					{
+						type: "forward-internal",
+						config: {
+							url: "https://${YOUR_API_DOMAIN}.internal",
+							binding: "internal",
+						},
+					},
+				],
+			},
+			{
+				actions: [
+					{
+						type: "forward-internal",
+						config: {
+							url: "https://${YOUR_HOMEPAGE_DOMAIN}.internal",
+							binding: "internal",
+						},
+					},
+				],
+			},
+		],
+	}}
+/>
+
+Our traffic policy simply forwards traffic to one of our two internal endpoints based on the path.
+More complex traffic handling can be created at your discretion.
+
+## **Step 5** — Create a ngrok Cloud Endpoint
+
+Now we can create a ngrok Cloud Endpoint that points to our
+newly created internal endpoint. Cloud Endpoints can be created
+via the API or ngrok dashboard.
+
+<Tabs groupId="guide">
+<TabItem value="api" label="API" default>
+We use the URL from Step 2 as the `url` value in the traffic policy, and the traffic policy from Step 3 as the `traffic-policy` value.
+The binding is set to public so anyone can access our cloud endpoint.
+
+```bash
+ngrok api endpoints create \
+    --api-key YOUR_NGROK_API_KEY \
+    --bindings public \
+    --description "sample description" \
+    --metadata sample-metadata \
+    --url https://${NGROK_SUBDOMAIN}.ngrok.app
+    --traffic-policy '{"on_http_request":[{"expressions":["req.url.path.startsWith('/api/')"],"actions":[{"type":"forward-internal","config":{"url":"https://{YOUR_API_DOMAIN}.internal"}}]},{"actions":[{"type":"forward-internal","config":{"url":"https://{YOUR_HOMEPAGE_DOMAIN}.internal"}}]}]}'
```

Comment:
```suggestion
    --traffic-policy "$(cat traffic-policy.yml)"
```

Or:

```suggestion
--traffic-policy "$(cat <<EOF
on_http_request:
# etc etc
EOF
)"
```

We really need to make `api` take a `--traffic-policy-file` flag too.